doc/releases/changelog-dev.md

```diff
@@ -138,10 +138,6 @@
 
 <h3>Improvements</h3>
 
-* Most compilation transforms, and relevant subroutines, have been updated to
-  support just-in-time compilation with `jax.jit`.
-  [(#1894)](https://github.com/PennyLaneAI/pennylane/pull/1894/)
-
```

Comment:
This was released with `v0.22.2` https://github.com/PennyLaneAI/pennylane/pull/2397 (and intended with `v0.22.0`), hence the changelog entry is redundant here.